docs/design/gh-ost-integration.md

```diff
@@ -0,0 +1,536 @@
+
+Streamer reads data from binary logs and streams it on.
+
+Applier connects and writes the applier-server, which is the server where migration happens. Applier is the one to actually write row data and apply binlog events onto ghost table.
+
+Throttler collects metrics related to throttling and makes informed decision whether throttling should take place. Here throttling means to pause.
+
+#### Requirements and limitations
+
+- `gh-ost` currently requires MySQL versions 5.6 and greater.
+- You will need to have one server serving Row Based Replication (RBR) format binary logs. Right now `FULL` row image is supported. `MINIMAL` to be supported in the near future. `gh-ost` prefers to work with replicas. You may still have your primary configured with Statement Based Replication (SBR).
+- If you are using a replica, the table must have an identical schema between the primary and replica.
+- `gh-ost` requires an account with these privileges:
+  - `ALTER, CREATE, DELETE, DROP, INDEX, INSERT, LOCK TABLES, SELECT, TRIGGER, UPDATE` on the database (schema) where your migrated table is, or of course on `*.*`
+  - either:
+    - `SUPER, REPLICATION SLAVE` on *.*, or:
+    - `REPLICATION CLIENT, REPLICATION SLAVE` on *.*
+The `SUPER` privilege is required for `STOP SLAVE, START SLAVE` operations. These are used on:
+- Switching your `binlog_format` to `ROW`, in the case where it is not ROW and you explicitly specified `--switch-to-rbr`
+  - If your replication is already in RBR (binlog_format=ROW) you can specify --assume-rbr to avoid the STOP SLAVE/START SLAVE operations, hence no need for SUPER.
+- Running --test-on-replica: before the cut-over phase, gh-ost stops replication so that you can compare the two tables and satisfy that the migration is sound.
+- Foreign key constraints are not supported.
+- Triggers are not supported.
+Different cloud providers require different flags.
+- Google Cloud SQL works, `--gcp` flag required.
+- Aliyun RDS works, `--aliyun-rds` flag required.
+- Azure Database for MySQL works, `--azure` flag required.
+
+### Goals
+
+We will enhance this feature gradually following the plan listed below in Milestone.
+For the first iteration, we will implement
+
+- Use gh-ost to execute migration
+  - Support "connect to primary, migrate on primary" only
+
+### Milestone
+
+1.
+    - Support "connect to primary, migrate on primary".
+    - Support non-tenant mode.
+    - Support on-premises MySQL databases.
+    - Divide gh-ost execution into stages. Proceed upon user's approval.
+1.
+    - Support gh-ost status feedback(ETA, configurations etc.)
+    - Improve migration logic to support rerun task.
+    - Support to reconfigure on the go.
+1. Support cloud MySQL databases.
+1. Support tenant mode.
+1. Support "connect to replica".
+
+## User workflow
+
+User can set a workspace-level gh-ost configuration, which will be the default.
```

Comment:
Unless we have confidence knowing which settings we allow users to set, otherwise, we can just display the settings we use, and do not allow them to change at first.

Reply:
Yes, we use gh-ost default settings for now. The design doc user workflow is what a full-fledged version would be.